docs/F5-vs-HAProxy.md

```diff
@@ -0,0 +1,43 @@
+# F5 or HAProxy Decision
+
+This document contains the information used to decide whether to leverage the built-in OpenShift HAProxy router as the default cluster ingress for HTTP(S) or the F5 BigIP Controller operator.
+
+## Background
+
+The F5 BigIP Controller operator was originally identified as the preferred tool to enable non-HTTP ingress to the OCP4 clusters.  The default OpenShift HAProxy routers do not support this type of traffic, and the integration with the F5 controller will enable automation and consistent implementation of these (non-HTTP) ingress types.
+
+The F5 BigIP Controller has additional functionality that would allow us to *replace* the OpenShift HAProxy routers all together.
+
+![Diagram A](images/F5-HAProxy-default-ingress.png)
+
+## Evaluation Criteria
+
+With duplicate functionality between the F5 and the HAProxy routers, the following points were used to compare and ultimately decide on the current path forward.
+
+### Route Features
+
+Both the F5 and the HAProxy routers support the same base Route configurations (Unsecured, Edge, Passthrough, Re-encrypt) and have a comparable feature set for basic router configuration (whitelist, load balance method, etc)
+
+The Features of the F5 router will allow for exposure of more complex load balancing features.  The majority of the current OCP deployed applications do not implement advanced load-balancing or router configuration at this time.
+
+### Performance
+
+Using the F5 as the default ingress route would remove 1 hop from all of the ingress to the platform and improve latency to running applications by a small amount.  The F5 appliances can also provide improved SSL encrypt/decrypt speeds, and offload the SSL processing (edge and re-encrypt) from the OCP Cluster.  Concerns have been identified with the projected amount of core/ram usage of the cluster and the current amount of free capacity in the Hosted F5 solution.  While it is not expected to be a performance bottleneck, it does identify the Hosted F5 service capacity as a critical external dependency.
+
```

Comment:
Should we also put in something to discuss the mechanism that does this? The F5 Operator needs to be up and running to push config changes into the F5 load balancer. It also has to go across a network to connect to that API. 

What about the partitioning? Do they segment off a partition per cluster? or is it per namespace?

Reply:
> What about the partitioning?

Yes, the CIS creates a partition for itself to use. Per cluster. We will be using separate VCMPs for each cluster as well to prevent any cross cluster issues or the CIS bleeding into other leveraged tenants.

Reply:
Was there a suggested change that we need to incorporate here?

Reply:
I don't think we need to add anything here about the mechanism for pushing config to the F5.